### PR TITLE
Registrations signed extension filtering

### PR DESCRIPTION
**Pull request by JohnReedV**
This PR implements Signed Extension registration filtering.

### CODE REVIEW

File: pallets/subtensor/src/lib.rs [hunk 1]
```diff
@@ -1825,10 +1825,20 @@ where
                 priority: Self::get_priority_vanilla(),
                 ..Default::default()
             }),
-            Some(Call::register { .. }) => Ok(ValidTransaction {
-                priority: Self::get_priority_vanilla(),
-                ..Default::default()
-            }),
+            Some(Call::register { netuid, .. }) => {
```
Comment (distributedstatemachine) [inline pallets/subtensor/src/lib.rs]: can we also add this for `burned_register` ? We should be able to use `||` in the method

File: pallets/subtensor/tests/registration.rs [hunk 2]
```diff
@@ -153,6 +154,99 @@ fn test_registration_ok() {
...
+            hotkey: hotkey_account_id,
+            coldkey: coldkey_account_id,
+        };
+        let info: DispatchInfo =
+            DispatchInfoOf::<<Test as frame_system::Config>::RuntimeCall>::default();
+        let extension = SubtensorSignedExtension::<Test>::new();
+        let result = extension.validate(&who, &call.into(), &info, 10);
+
+        // Expectation: The transaction should be rejected
+        assert!(result.is_err());
```
Comment (distributedstatemachine) [inline pallets/subtensor/tests/registration.rs]: Do you think it would be a good idea to asser the actual error ?

**Review by distributedstatemachine — APPROVED** [review]

**Review by sam0x17 — APPROVED** [review]
looks sound to me, and tests look thorough

**Comment by distributedstatemachine** [comment]
LGTM!

**Pull request merged**